### ISSUE TITLE
[Question] Theme template for Gutenberg (starter theme)

### ISSUE THREAD

**Issue by mahnunchik** [WordPress/gutenberg#21401]
> I think it is right place to ask such question because WordPress/WordPress repository has disabled issues.

It would be nice to have starter theme with Gutenberg support for developers. It should simplify and standardize theme development.

Some investigation:

* [Theme Development Examples](https://developer.wordpress.org/themes/getting-started/theme-development-examples/) — lack of information about Gutenberg ready themes.
* [_s or underscores](https://github.com/Automattic/_s) — it seems abandoned.
* https://github.com/WordPress/twentynineteen — archived and doesn't receive updates. Lack of documentation about building tools.
* https://github.com/WordPress/theme-experiments — interesting but marked to not ready for production use.
* https://github.com/Automattic/themes — lack of information about Gutenberg ready themes.
* https://github.com/Automattic/newspack-theme — may it be used as template?
* [roots/sage](https://github.com/roots/sage) — requires Blade templating engine, custom file structure, and another custom things.

Additional questions:

* Is Full site editing feature dead?

**Comment by maheshwaghmare** [comment]
> It would be nice to have starter theme with Gutenberg support for developers. It should simplify and standardize theme development.

Right, We need a such Gutenberg ready starter theme. I think for now we can use the `_s` theme as a starter theme from [scaffold ](https://developer.wordpress.org/cli/commands/scaffold/) command or from the https://github.com/Automattic/_s.

And to add the Gutenberg support we can check the handbook [theming for the block editor](https://developer.wordpress.org/block-editor/developers/themes/).

> Is Full site editing feature dead?

Nope, The Full site editing feature is listed into the [Phase 2](https://github.com/WordPress/gutenberg/issues/13113). We can see the full site editing experiment details at https://github.com/WordPress/gutenberg/issues/20791

I think once the full site editing experiment complete and it is ready to merge into the core then it is integrated into the the theme scaffold command or we have something new package like https://github.com/WordPress/gutenberg/pull/19773 which allow use to create a Gutenberg ready theme.

> Theme Development Examples — lack of information about Gutenberg ready themes.

I think yes, Because of the full site editing is still in development. It use the `block-template-parts` and `block-templates` to render the HTML files which render the static and dynamic blocks.

In the handbook the section [template files](https://developer.wordpress.org/themes/basics/template-files/) shows that to create PHP files for the templates like:

- header.php
- singular.php
- single.php
- etc

But, In full site editing the HTML templates are used to rendering the static & dynamic content. So, The templates will be like:

- header.html
- singular.html
- single.html
- etc

For now it is not clear that how templates & template parts are integrate and what about the PHP template files. There are lot of questions about implementing the full site experiment. For that the documentation is not updated yet.

You can track [all issues](https://github.com/WordPress/gutenberg/issues?q=is%3Aissue+is%3Aopen+label%3A%22%5BFeature%5D+Full+Site+Editing%22) of full site editing and [all PRs](https://github.com/WordPress/gutenberg/pulls?q=is%3Apr+is%3Aopen+label%3A%22%5BFeature%5D+Full+Site+Editing%22).

Also, I recommend you to see all the closed issues & closed PRs to see what was happening into the full site editing feature.

> https://github.com/Automattic/newspack-theme — may it be used as template?

I think there is no difference from the `_s` and the `newspack-theme`. From the repository, there is no such template system which render the HTML templates from the Gutengerg.

**Comment by mahnunchik** [comment]
Hi @maheshwaghmare thank you for the clarification.

I will use `_s` as template for now and I'm looking forward for Gutenberg ready starter theme.

**Comment by maheshwaghmare** [comment]
Yup! I'm also excited about the full site editing.

**Comment by HannesDev** [comment]
> _s or underscores — it seems abandoned.

FYI: @Ismail-elkorchi is currently heavily updating `_s`. The agenda is not quite clear to be but I guess Gutenberg support becomes a essential part.

**Comment by carolinan** [comment]
It has been very difficult to build a block starter theme because of how frequent the blocks, block CSS and markup changes.
We are in a better position now than we have been ever before. But it is still a matter of who is willing to write a theme, **that is going to have to be rewritten over and over and over again?** As much as one wants to help the community, it is something that requires a lot of time and dedication.

**Comment by talldan** [comment]
Closing as I can't see any action items from this, but feel free to continue the conversation.

**Comment by mahnunchik** [comment]
😢 😭  Looking for best practices for modern theme development 😉
